lib/widgets/profile.dart

```diff
@@ -0,0 +1,272 @@
+    if (userFields.isNotEmpty) {
+      return DefaultTextStyle.merge(
+        style: const TextStyle(fontSize: 16, height: 1),
+        textAlign: TextAlign.left,
+        child: Column(
+          children: [
+            const SizedBox(height: 16),
+            Table(
+              border: TableBorder.all(style: BorderStyle.none),
+              columnWidths: const {
+                0: FixedColumnWidth(116),
+                1: FlexColumnWidth()},
+              defaultVerticalAlignment: TableCellVerticalAlignment.top,
+              children: userFields.map((e) => TableRow(
+                children: [
+                  Padding(
+                    padding: const EdgeInsets.fromLTRB(0, 4, 8, 4),
+                    child: Text(e.label, style: const TextStyle(fontWeight: FontWeight.bold, ))),
+                  renderProfileDataField(context, e, store),
+                ])).toList()),
+          ]));
+    } else {
+      return const SizedBox.shrink();
+    }
+  }
+}
+
+class ProfilePage extends StatelessWidget {
+  final int userId;
+
+  const ProfilePage({super.key, required this.userId});
+
+  static Route<void> buildRoute({required BuildContext context, required int userId}) {
+    return MaterialAccountWidgetRoute(context: context,
+      page: ProfilePage(userId: userId));
+  }
+
+  Widget buildErrorPage() {
+    return Scaffold(
+      appBar: AppBar(title: const Text("Error")),
+      body: const SingleChildScrollView(
+        child: Padding(
+          padding: EdgeInsets.all(16),
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              Icon(Icons.error),
+              Text("Could not show user profile."),
+            ]))));
+  }
+
+  void _createDmNavigation (context) {
+    final store = PerAccountStoreWidget.of(context);
+    final allRecipientIds = [store.account.userId, userId];
+    allRecipientIds.sort();
+    final narrow = DmNarrow(
+      selfUserId: store.account.userId,
+      allRecipientIds: allRecipientIds);
+    Navigator.push(context,
+      MessageListPage.buildRoute(context: context, narrow: narrow));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final user = store.users[userId];
+    if (user != null) {
+      final avatar = Avatar(userId: userId, size: 200, borderRadius: 200 / 8);
+      final Widget profileDataTable;
+      if (user.profileData != null) {
+        profileDataTable = ProfileDataTable(user:user);
+      } else {
+        profileDataTable = const SizedBox.shrink();
+      }
+      final Widget localTimeWidget;
+      if (!user.isBot) {
+        final userLocalNow = getNowInTimezone(user.timezone);
+        if (userLocalNow != null) {
+          localTimeWidget = Text("${DateFormat.jm().format(userLocalNow)} Local time",
+            style: const TextStyle(fontWeight: FontWeight.normal));
```

Comment:
This and the rest of the `localTimeWidget` logic can probably best be moved into a little widget of their own, as the work they're doing is pretty independent of the rest of this method.

As a bonus, that would set us up to make an enhancement where that widget would rebuild every time the answer changes (so at the start of each minute), instead of getting out of date if the screen stays open for a while and then you come back to it. That'd be neat. (Wouldn't be worth tons of effort; but my guess is that it's pretty straightforward once one lays hands on the right API.)